utils/local-engine/Shuffle/ShuffleSplitter.cpp

```diff
@@ -249,7 +249,28 @@ void ColumnsBuffer::add(DB::Block & block, int start, int end)
     }
     assert(!accumulated_columns.empty());
     for (size_t i = 0; i < block.columns(); ++i)
-        accumulated_columns[i]->insertRangeFrom(*block.getByPosition(i).column, start, end - start);
+	{
+        // for union, the columns type may be different for the two steam, one is nullable, the other not.
+        std::string l_name = typeid(*accumulated_columns[i]).name();
+        std::string r_name = typeid(*block.getByPosition(i).column).name();
```

Comment:
it is better to use checkAndGetColumn instead of typeid().name()

Reply:
auto l_type = accumulated_columns[i]->getDataType();
auto r_type = block.getByPosition(i).column->getDataType();

if (l_type == r_type)
{
   xxx
}
else if (l_type == TypeIndex::Nullable)
{
  xxx
}
else if (r_type == TypeIndex::Nullable)
{
  xxx
}